### PR TITLE
CMR-7073 Adding in Direct Distribution Policy to UMM-C, ECHO10, DIF10 and translate between them and ISO.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/echo10.clj [hunk 1]
```diff
@@ -222,4 +222,11 @@
...
      (spatial/spatial-element c)
-     (ru/generate-browse-urls (:RelatedUrls c))]))
+     (ru/generate-browse-urls (:RelatedUrls c))
+     (when-let [direct-dist-info (:DirectDistributionInformation c)]
+       [:DirectDistributionInformation
+         [:Region (:Region direct-dist-info)]
+         (for [prefix-name (:S3BucketAndObjectPrefixNames direct-dist-info)]
+           [:S3BucketAndObjectPrefixName prefix-name])
+         [:S3CredentialsAPIEndpoint (:S3CredentialsAPIEndpoint direct-dist-info)]
+         [:S3CredentialsAPIDocumentationURL (:S3CredentialsAPIDocumentationURL direct-dist-info)]])]))
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/echo10.clj:232]: This looks exactly the same as the `dif10` mapping. Is it worth having a common function between the two formats? Not sure how likely it is that they would diverge

File: umm-spec-lib/test/cmr/umm_spec/test/umm_to_xml_mappings/dif10.clj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                    :S3CredentialsAPIDocumentationURL "https://www.credAPIDocURL.org"}})
+        diff-result (first (select result "/DIF/DirectDistributionInformation"))]
+    (is (and
+          (= "us-west-1" (value-of diff-result "Region"))
+          (= "bucket1" (first (values-at diff-result "S3BucketAndObjectPrefixName")))
+          (= "bucket2" (second (values-at diff-result "S3BucketAndObjectPrefixName")))
+          (= "https://www.credAPIURL.org" (value-of diff-result "S3CredentialsAPIEndpoint"))
+          (= "https://www.credAPIDocURL.org" (value-of diff-result "S3CredentialsAPIDocumentationURL"))))))
+
+(deftest dif10-direct-distribution-information-nil-test
```
Comment (mschmele) [inline umm-spec-lib/test/cmr/umm_spec/test/umm_to_xml_mappings/dif10.clj:25]: In addition to `nil` cases, I'd like to see at least one case with incorrect metadata for these tests in each format

**Review by mschmele — COMMENTED** [review]
Looks pretty good to me - I'll approve once you add a test case with incorrect data for the mapping tests

File: system-int-test/test/cmr/system_int_test/search/umm_json_search_test.clj [hunk 3]
```diff
@@ -96,7 +96,7 @@
         "1.0" nil "umm_json_v1_0"
 
         "Retrieve specified version 1.15.4 with URL extension"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/umm_json_search_test.clj:98]: update the comment

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/distributions_related_url.clj [hunk 4]
```diff
@@ -191,7 +192,7 @@
                               (char-string-value url "gmd:description"))
               service-url (first (filter #(= url-link (:URL %)) service-urls))
               type (or opendap-type (:Type types-and-desc) (:Type service-url) "GET DATA")]
-        :when (seq type)]
+        :when (not (= href-type "DirectDistributionInformation"))]
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/distributions_related_url.clj:195]: This logic looks wrong.

**Review by ygliuvt — APPROVED** [review]

**Review by mschmele — APPROVED** [review]

**Pull request merged**